https://bz-attachments.freebsd.org/attachment.cgi?id=204611
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=238128

Stripping RPATH on Prefix is really bad, because it means the default
RPATH entries disappear, and any installed binaries don't work (get
kernel traps).

Extracted the minstall patch from the FreeBSD full patch.

--- a/mesonbuild/minstall.py
+++ b/mesonbuild/minstall.py
@@ -488,8 +488,11 @@
                         printed_symlink_error = True
             if os.path.isfile(outname):
                 try:
-                    depfixer.fix_rpath(outname, install_rpath, final_path,
-                                       install_name_mappings, verbose=False)
+                    if install_rpath:
+                        depfixer.fix_rpath(outname, install_rpath, final_path,
+                                           install_name_mappings, verbose=False)
+                    else:
+                        print("RPATH changes at install time disabled")
                 except SystemExit as e:
                     if isinstance(e.code, int) and e.code == 0:
                         pass
